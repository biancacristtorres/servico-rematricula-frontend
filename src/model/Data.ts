export class Data {
  public data!: Date;
  public horaInicio!: Date;
  public horaFim!: Date;
}
