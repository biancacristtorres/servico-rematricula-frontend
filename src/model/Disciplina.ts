import { SituacaoDaDisciplina } from '@/common/enum/SituacaoDaDisciplina';
import { Turma } from './Turma';

export class Disciplina {
  public codigo: number;
  public nome: string;
  public situacao: SituacaoDaDisciplina;
  public modulo: string;
  public eDisciplinaEletiva: boolean;
  public possuiEquivalenciaDoisPorUm!: boolean;
  public turmas: Turma[] = [];
  public turmasEquivalentes: Turma[] = [];
  public turmasEspeciais: Turma[] = [];

  constructor(codigo: number, nome: string, situacao: SituacaoDaDisciplina,
              modulo: string, eDisciplinaEletiva: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.situacao = situacao;
    this.modulo = modulo;
    this.eDisciplinaEletiva = eDisciplinaEletiva;
  }
}
