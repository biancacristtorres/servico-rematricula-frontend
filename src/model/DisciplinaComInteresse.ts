export class DisciplinaComInteresse {
  public codigo?: number;
  public nome?: string;
  public interesseManifestado?: boolean;

  constructor(codigo: number, nome: string, interesseManifestado: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.interesseManifestado = interesseManifestado;
  }
}
