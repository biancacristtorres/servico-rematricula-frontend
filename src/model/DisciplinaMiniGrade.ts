export class DisciplinaMiniGrade {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
}
