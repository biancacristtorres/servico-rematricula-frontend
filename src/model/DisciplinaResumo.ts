import { StatusDaDisciplina } from '@/common/enum/StatusDaDisciplina';

export class DisciplinaResumo {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
  public descricaoTurma?: string;
  public status?: StatusDaDisciplina;
  public ePorData?: boolean;
  public campus?: string;
  public descricaoTipoCargaHoraria?: string;
  public datas?: string[];
  public nomeDisciplinaEquivalente?: string;
}
