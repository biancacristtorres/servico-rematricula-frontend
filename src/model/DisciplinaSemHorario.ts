export class DisciplinaSemHorario {
  public nomeDisciplina?: string;
  public nomeDisciplinaEquivalente?: string;
}
