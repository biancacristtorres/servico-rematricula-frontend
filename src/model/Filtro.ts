export class Filtro {
  public exibirTurnosDiferentes: boolean;

  constructor() {
    this.exibirTurnosDiferentes = true;
  }
}
