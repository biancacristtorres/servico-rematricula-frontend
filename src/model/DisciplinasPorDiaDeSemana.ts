import { DisciplinaResumo } from './DisciplinaResumo';

export class DisciplinasPorDiaDeSemana {
  [index: number]: DisciplinaResumo[];
  public 2: DisciplinaResumo[];
  public 3: DisciplinaResumo[];
  public 4: DisciplinaResumo[];
  public 5: DisciplinaResumo[];
  public 6: DisciplinaResumo[];
  public 7: DisciplinaResumo[];
}
