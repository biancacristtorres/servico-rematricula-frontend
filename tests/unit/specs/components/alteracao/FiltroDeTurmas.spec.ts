import { Filtro } from '@/model/Filtro';
import {
  FiltroDeTurmasFactory,
} from '../../../factory/components/alteracao/FiltroDeTurmasFactory';

describe('Componente de Filtro de Turmas', () => {
  const factory = new FiltroDeTurmasFactory();

  it('Ao clicar em aplicar filtro, um novo filtro Ã© aplicado na store', async (done) => {
    const filtro = new Filtro();
    filtro.exibirTurnosDiferentes = true;
    factory.dadoOModel(filtro);

    await factory.montarComponente();
    factory.aplicarFiltro();

    expect(factory.filtroDaStoreDeAlteracao.exibirTurnosDiferentes).toBeTruthy();
    done();
  });

  it('Snapshot', async (done) => {
    await factory.montarComponente();

    expect(factory.componente).toMatchSnapshot();
    done();
  });
});
