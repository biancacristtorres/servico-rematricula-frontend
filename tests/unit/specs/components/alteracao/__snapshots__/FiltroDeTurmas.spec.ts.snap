// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Filtro de Turmas Snapshot 1`] = `
<div class="v-menu v-menu--inline">
  <div class="v-menu__activator"><button type="button" class="blue--text v-btn v-btn--flat theme--light" id="exibir-filtro">
      <div class="v-btn__content"><i aria-hidden="true" class="v-icon v-icon--left 
      mdi 
      mdi-filter
     theme--dark blue--text"></i>
        Filtrar
      </div>
    </button></div>
  <div class="v-menu__content theme--light " style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;">
    <div class="v-card theme--light">
      <div class="v-card__text">
        <div class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light">
          <div class="v-input__control">
            <div class="v-input__slot" style="height: auto;">
              <div role="radiogroup" class="v-input--radio-group__input"><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">
                  <div class="primary--text">
                    Turno:
                  </div>
                </label>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" role="radio" type="radio" name="v-radio-29" value="false">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">radio_button_unchecked</i>
                  </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">
                    <div>
                      Apenas no meu turno
                    </div>
                  </label>
                </div>
                <div class="v-radio theme--light accent--text">
                  <div class="v-input--selection-controls__input"><input aria-checked="true" role="radio" type="radio" name="v-radio-29" value="true">
                    <div class="v-input--selection-controls__ripple accent--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light accent--text">radio_button_checked</i>
                  </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">
                    <div>
                      Todos turnos dispon√≠veis
                    </div>
                  </label>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
          </div>
        </div>
      </div>
      <div class="v-card__actions">
        <div class="spacer"></div> <button type="button" class="v-btn v-btn--flat theme--light">
          <div class="v-btn__content">Fechar</div>
        </button> <button type="button" class="v-btn v-btn--flat theme--light primary--text" id="aplicar-filtro">
          <div class="v-btn__content">
            Aplicar
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
